### PR TITLE
Responding mentor comments

### PR DESCRIPTION
**Pull request by soft-circles**

### CODE REVIEW

File: game/game_controller.rb [hunk 1]
```diff
@@ -90,6 +99,16 @@ def play
     game_end
   end
 
+  def game_type
+    if players.all? { |player| player.class == Human }
+      "HvH"
```
Comment (td-extreme) [inline game/game_controller.rb]: I would suggest making the game_type an enum instead of having magic strings.

File: game/game_controller.rb [hunk 2]
```diff
@@ -76,9 +85,9 @@ def display
 
   def take_turns
     players.each do |player|
-      display
+      # display
```
Comment (td-extreme) [inline game/game_controller.rb]: You should NOT commit commented out code. Just delete it. We have version control so you can always get this back by looking in the commit logs.

File: app.rb [hunk 3]
```diff
@@ -12,14 +12,15 @@
 end
 
 get '/game' do
-  game = session[:game]
-  if game
-    @board = game.board.spaces.each_slice(3).to_a
-    @game = game
-    erb :game
+  if session[:game]
```
Comment (td-extreme) [inline app.rb:15]: :+1:

File: spec/app/app_spec.rb [hunk 4]
```diff
@@ -29,12 +29,18 @@
       follow_redirect!
       expect(last_request.path).to eq('/')
     end
+
+    it "game should start with computer move if computer player going first" do
```
Comment (td-extreme) [inline spec/app/app_spec.rb:33]: :+1:

**Review by td-extreme — COMMENTED** [review]
I just looked over the Ruby code. Looks good to me for the most part.

File: game/game_controller.rb [hunk 5]
```diff
@@ -53,6 +54,14 @@ def active_player
     end
   end
 
+  def inactive_player
```
Comment (td-extreme) [inline game/game_controller.rb]: You have a attr_accessor defined for `inactive_player` on line six. Does this function override that?

File: game/game_controller.rb [hunk 6]
```diff
@@ -53,6 +54,14 @@ def active_player
...
+    if board.empty_spaces.odd?
```
Comment (td-extreme) [inline game/game_controller.rb]: When making a simple if else like this you can use the : ? ternary operator. 
```

def inactive_player 
  board.empty_spaces.odd? ? players[1] : players[2]
end

```
https://en.wikipedia.org/wiki/%3F:#Ruby

File: game/game_controller.rb (hunk 1 above)
Comment (td-extreme) [reply game/game_controller.rb]: Also, who calls this method? Why do we need to be able to know what the game type is?

File: public/application.js [hunk 7]
```diff
@@ -15,62 +14,45 @@ function humanVersusComputer() {
...
+    });
   });
 };
 
-function makeHumanMove(response) {
-  if (response.valid === true) {
-    $('#' + response.spot).empty();
-    $('#' + response.spot).append(response.marker);
-    endGame(response);
+function gameIsOver(response) {
```
Comment (TakaGoto) [inline public/application.js]: is this function necessary? if we rename response object to something like `response.isGameOver` we wouldn't need this function.

File: public/application.js [hunk 8]
```diff
@@ -1,9 +1,8 @@
 $(document).ready(() => {
-  humanVersusHuman();
-  humanVersusComputer();
+  play();
 });
 
-function humanVersusComputer() {
+function play() {
   $("td").click(function(e) {
```
Comment (TakaGoto) [inline public/application.js]: use the es6 anonymous function syntax instead, aka `(e) => {}`

File: public/application.js [hunk 9]
```diff
@@ -15,62 +14,45 @@ function humanVersusComputer() {
...
-        },
-        dataType: "json"
-    });
-    request.then((response) => {
-        makeHumanMove(response);
-        endGame(response);
-      });
+function putMarkersDown(response) {
+  response.moves.forEach(move => {
+    if (move.valid == true) {
```
Comment (TakaGoto) [inline public/application.js]: always use triple  equals in javascript: https://stackoverflow.com/questions/523643/difference-between-and-in-javascript

File: public/application.js (hunk 8 above)
Comment (td-extreme) [reply public/application.js]: :+1:

File: public/application.js (hunk 9 above)
Comment (TakaGoto) [reply public/application.js]: also, what should happen if the move isn't valid? or can the move ever not be valid?

File: public/application.js [hunk 10]
```diff
@@ -15,62 +14,45 @@ function humanVersusComputer() {
...
+      $('#' + move.spot).empty();
```
Comment (TakaGoto) [inline public/application.js]: lets use the nicer string interpolation, `#${move.spot}`

**Comment by TakaGoto** [comment]
LGTM. where are the javascript tests?

**Pull request merged**